alibuild_helpers/build.py

```diff
@@ -72,37 +71,84 @@ def syncToLocal(self, p, spec):
   def syncToRemote(self, p, spec):
     pass
 
+class CurlError(Exception):
+  def __init__(self, rv):
+    self.rv = rv
+  def __str__(self):
+    return "curl execution returned %d" % self.rv
+
 class HttpRemoteSync:
   def __init__(self, remoteStore, architecture, workdir, insecure):
     self.remoteStore = remoteStore
     self.writeStore = ""
     self.architecture = architecture
     self.workdir = workdir
     self.insecure = insecure
+    self.httpTimeoutSec = 15
+    self.httpConnRetries = 4
+    self.httpBackoff = 0.4
+    self.doneOrFailed = []
+
+  def getRetry(self, url, dest=None):
```

Comment:
Why not using `curl` `--retry*` options? It's there since `7.12` and even `slc6` has `7.19`.

Reply:
Yes, I am aware - we're using those options already in the publisher scripts for instance. We could use `--retry <nretries> --max-time <timeout> --retry-delay <backoff>`. I don't have a strong opinion, the only point being that `curl` does not support either fractions of seconds in backoff or an incremental backoff.